whitepaper/interledger.tex

```diff
@@ -0,0 +1,351 @@
+now  owes an additional {\tt amount} units of value to Bob.
+
+\subsection{Interledger Transfers}
+An Interledger Transfer is a specific kind of Hash Time Lock Agreement, designed to form multi-transfer chains of HTLAs in the network. To this end,
+it specifies a {\tt destination}, which is an ILP address: a hierarchical identifier for an agent in the network who may be able to produce the SHA256 preimage of the
+HTLA's hash. Apart from this destination field, the packets that describe an Interledger transfer also have {\tt data} fields, which should be transported end-to-end over a multi-transfer chain (see below).
+
+\section{Multi-transfer Hashlocks}
+Suppose Alice sends an Interledger Transfer to Bob, whose {\tt destination} points at one of Bob's other network neighbors, Charlie. Bob can then create a second
+transfer, from him to Charlie, with the same {\tt hash}, and with a slightly earlier {\tt time} deadline. If Charlie fulfills this transfer from Bob on time, then Bob will be able
+to use the preimage presented by Charlie, to fulfill the incoming transfer from Alice before its deadline. We call this a two-transfer payment (from Alice, via Bob, to Charlie) and the same
+chaining method can be repeated to create payments with more than two transfers.
+
+\section{ILP Addresses}
+As stated earlier, an ILP address is a hierarchical identifier for a specific agent in the network.
+An ILP address is a string, consisting of one or more segments, separated by dots ({\tt .}). Each segment may contain alphanumeric characters (upper or lower case, case-sensitive),
+underscores ({\tt \_}),
+tildes ({\tt \textasciitilde}),
+and hyphens ({\tt -}).
+
+\subsection{Static Routing}
+In Interledger networks where the set of agents participating in the network and the connections between them don't change,
+an easy way to make sure that all agents in the network know how to route multi-transfer payments towards their destination, would be to assign an ILP address to each agent, and statically configure the forwarding rules for each destination in each agent. In any case, the first two agents to form a new network can choose their own ILP addresses that way.
+But when a new agent joins, one of two things can happen:
+they may become a child of the agent through which they get joined to the network, or become a first-class citizen.
+
+\subsection{Sub-Addresses}
+If the new agent becomes a child, then it takes the address of the parent through which it joins the network, and adds one or more segments at the end. By using its parent's address as a
+prefix, other agents in the network will always be able to route payments to the new agent, provided they know how to route a payment to the new agent's parent. This is the hierarchical
+way to generate new ILP addresses.
+
+\subsection{Route Broadcasts}
```

Comment:
I think we need to model the case of conflicting broadcasts more thoroughly and be more prescriptive about the behaviour of a node that receives conflicting broadcasts.

Reply:
how would you distinguish conflict (so address collision, two destinations call themselves by the same name) from the (desirable!) situation where there are two or more alternative routes to the same destination?

Reply:
cc @justmoon